framework/base/Security.php

```diff
@@ -305,7 +305,7 @@ public function hkdf($algo, $inputKey, $salt = null, $info = null, $length = 0)
         if (is_string($length) && preg_match('{^\d{1,16}$}', $length)) {
             $length = (int) $length;
         }
-        if (!is_integer($length) || $length < 0 || $length > 255 * $hashLength) {
+        if ($length < 0 || $length > 255 * $hashLength || !is_integer($length)) {
```

Comment:
Good for you, maybe I'm not such a gifted person. I read that line as "$length is out of bounds or not an integer at all... wait, so why I'm comparing it in the first place?". That's [lower code quality](http://i.imgur.com/J1svNp7.jpg) in my perspective.